adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m

```diff
@@ -13,31 +13,34 @@
 // limitations under the License.
 
 #import "GADMAdapterUnityBannerAd.h"
-
 #import "GADMAdapterUnityConstants.h"
-#import "GADMAdapterUnitySingleton.h"
 #import "GADMAdapterUnityUtils.h"
 #import "GADMediationAdapterUnity.h"
+#import "GADUnityError.h"
 
 @interface GADMAdapterUnityBannerAd () <UADSBannerViewDelegate>
 @end
 
 @implementation GADMAdapterUnityBannerAd {
   /// Connector from Google Mobile Ads SDK to receive ad configurations.
   __weak id<GADMAdNetworkConnector> _connector;
-
+  
   /// Adapter for receiving ad request notifications.
   __weak id<GADMAdNetworkAdapter> _adapter;
-
+  
   /// Unity Ads banner ad object.
   UADSBannerView *_bannerAd;
-
+  
   /// Unity ads placement ID.
   NSString *_placementID;
 }
 
-- (nonnull instancetype)initWithGADMAdNetworkConnector:(nonnull id<GADMAdNetworkConnector>)connector
-                                               adapter:(nonnull id<GADMAdNetworkAdapter>)adapter {
+- (instancetype)initWithGADMAdNetworkConnector:(nonnull id<GADMAdNetworkConnector>)connector
+                                       adapter:(nonnull id<GADMAdNetworkAdapter>)adapter {
+  if (![UnityAds isInitialized]) {
+    NSLog(@"Unity Ads Adapter Error: Unity Ads is not initialized.");
+    return nil;
+  }
```

Comment:
I think this check should go outside of this init call and in the respective `getBanner`/`getInterstitial` methods.
Additionally, the adapters should attempt to initialize the SDK if it has not been initialized yet.